### ISSUE TITLE
S3 Bucket Tier Configuration: `Days specified in ARCHIVE_ACCESS tier should not be less than 90`

### ISSUE THREAD

**Issue by BwL1289** [aws-cloudformation/cfn-lint#3992]
### CloudFormation Lint Version

cfn-lint 1.27.0

### What operating system are you using?

Mac

### Describe the bug

Cfnlint should notify users if they have an improper config for S3 bucket tiering configuration. For example, a configuration with the following yields an error when deployed: `Days specified in ARCHIVE_ACCESS tier should not be less than 90.`

```yaml
MyBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      IntelligentTieringConfigurations:
        - Id: ColdIntelligentTieringConfig
          Status: Enabled
          Tierings:
            - AccessTier: ARCHIVE_ACCESS
              Days: 1
            - AccessTier: DEEP_ARCHIVE_ACCESS
              Days: 7
```

### Expected behavior

Cfnlint raises an error when any s3 bucket tier configuration is incorrect.

### Reproduction template

```yaml
MyBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      IntelligentTieringConfigurations:
        - Id: ColdIntelligentTieringConfig
          Status: Enabled
          Tierings:
            - AccessTier: ARCHIVE_ACCESS
              Days: 1
            - AccessTier: DEEP_ARCHIVE_ACCESS
              Days: 7
```
